sharding/mainchain/interfaces.go

```diff
@@ -0,0 +1,48 @@
+package mainchain
+
+import (
+	"context"
+	"math/big"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/sharding/contracts"
+)
+
+// Signer defines an interface that can read from the Ethereum mainchain as well call
+// read-only methods and functions from the Sharding Manager Contract.
+type Signer interface {
+	Sign(hash common.Hash) ([]byte, error)
+}
+
+// ContractManager specifies an interface that defines both read/write
+// operations on a contract in the Ethereum mainchain.
+type ContractManager interface {
+	ContractCaller
+	ContractTransactor
+}
+
+// ContractCaller defines an interface that can read from a contract on the
+// Ethereum mainchain as well as call its read-only methods and functions.
+type ContractCaller interface {
+	SMCCaller() *contracts.SMCCaller
+	GetShardCount() (int64, error)
+	Account() *accounts.Account
```

Comment:
We don't really Account address from contract. At the same time, I'm not sure where else to put it. Do you think the Reader interface is more appropriate for `Account()`?